src/Shift-ClassBuilder-Tests/FluidClassBuilderTest.class.st

```diff
@@ -0,0 +1,243 @@
+	#superclass : #TestCase,
+	#instVars : [
+		'builder'
+	],
+	#category : #'Shift-ClassBuilder-Tests'
+}
+
+{ #category : #running }
+FluidClassBuilderTest >> builder [ 	
+
+	^  builder
+]
+
+{ #category : #'tests - mandatory' }
+FluidClassBuilderTest >> removeFromSystem [
+
+	self class environment at: #MyClass 
+		ifPresent: [:p | p removeFromSystem: false ]
+		ifAbsent: [  ] .
+	
+]
+
+{ #category : #running }
+FluidClassBuilderTest >> setUp [
+	super setUp.
+	builder := FluidClassBuilder new 
+		superclassToBuild: Object;
+		classNameToBuild: #Point33;
+		package: 'FakedPackage'
+
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuildLayout [
+
+	builder 
+		layout: VariableLayout;
+		package: 'Collections-Sequenceable'.
+	self assert: builder layoutToBuild equals: VariableLayout
+
+	
+]
+
+{ #category : #'tests - build' }
+FluidClassBuilderTest >> testBuildSimplePoint2 [
+
+	| clas |
+	builder := Object < #Point2.
+	builder slots: { #x . #y }.
+	builder package: 'FakedCore'.
+	clas := builder build.
+	
+	self assert: clas superclass equals: Object.
+	self assert: clas name equals: #Point2.
+	self assert: clas slots size equals: 2.
+	self flag: #askPablo.
+	"self assert: clas package packageName equals: 'FakedCore'"
+	
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuilderSharedPools [
+
+	builder slots: {#string . #runs};
+		tag: 'Base';
+		sharedPools: 'TextConstants';
+		package: 'Text-Core'.  
+	self assert: builder sharedPoolsToBuild equals: 'TextConstants'.
+	
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuilderSharedVariables [
+
+	builder 
+		layout: ByteLayout;
+		sharedVariables: { #NonAsciiMap };
+		package: 'Collections-Strings'.  
+	self assert: builder sharedVariablesToBuild equals: { #NonAsciiMap }.
+	self assert: builder layoutToBuild equals: ByteLayout.
+	
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuilderTraits [
+	
+	builder uses: TViewModel.
+	self assert: builder traitsToBuild equals: TViewModel 
+]
+
+{ #category : #'tests - mandatory' }
+FluidClassBuilderTest >> testBuilderWithPackage [
+
+	builder package: 'Kernel-BasicObjects'.
+	self assert: builder packageToBuild equals: 'Kernel-BasicObjects'
+
+	
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuilderWithSlots [
+
+	builder slots: { #x . #y }.
+	self assert: builder slotsToBuild equals: { #x . #y }.
+
+	
+]
+
+{ #category : #'tests - gathering' }
+FluidClassBuilderTest >> testBuilderWithTag [
+
+	builder tag: 'Foo'.
+	self assert: builder tagToBuild equals: 'Foo'.
+
+	
+]
+
+{ #category : #'tests - mandatory' }
+FluidClassBuilderTest >> testCreateBuilder [
+
+	builder := Object < #Point.
+	self assert: builder superclassToBuild equals: Object.
+	self assert: builder classNameToBuild equals: #Point.
+	
+]
+
+{ #category : #'tests - mandatory' }
+FluidClassBuilderTest >> testCreateBuilderWithNil [
+
+	builder := nil < #NewProtoObject.
```

Comment:
#< is really bad message for nil object. It will eat errors in many comparison cases: 
```Smalltalk
aNumber < 10
aString < 'anotherString'
```

Reply:
BTW I think that expecting that aString is nil in a comparison like <'anotherstring' is definively a bug so I do not understand why getting an error from the classbuilder is worse than now.
I will move this discussion to https://github.com/pharo-project/pharo/issues/7471